src/topology/mod.rs

```diff
@@ -453,6 +471,41 @@ impl RunningTopology {
             info!("Starting sink {:?}", name);
             self.spawn_sink(&name, &mut new_pieces);
         }
+
+        // API
+        #[cfg(feature = "api")]
+        if let Some(diff) = &diff.api {
+            self.spawn_api(diff, &mut new_pieces).await
+        }
+    }
+
+    #[cfg(feature = "api")]
+    /// Starts/restarts the API server, if enabled in configuration
+    async fn spawn_api(&mut self, diff: &api::Difference, new_pieces: &mut builder::Pieces) {
+        let message = match diff {
+            api::Difference::Start => "Starting",
+            api::Difference::Restart => "Restarting",
+            _ => return, // no effect if the server has stopped
+        };
+
+        let server = new_pieces.api.take().unwrap();
+
+        info!(
+            message = format!("{} API server", message).as_str(),
+            ip = server.ip().to_string().as_str(),
```

Comment:
This seems a little verbose, but `tracing::Value` is required on a value, so the usual `String` -> `&str` deref didn't seem to work.